File: src/download_data.R
# download_data.R
# author: Kangbo Lu, Mengyuan Zhu, Mitchie Zhao, Siqi Zhou
# date: 2020-11-28

"This script downloads the dataset and stores it in the designated location.

Usage: src/download_data.R --url=<url> --output=<output_file>
Options:
--url=<url>               Takes a url from where to download the data as a csv file
--output=<output_file>    Takes a local file path to store csv data
" -> doc

# import libraries
library(docopt)
library(RCurl)
opt <- docopt(doc)

# define main function
main <- function(opt) {
    # generate error message if the input url is not valid
    if (!url.exists(opt$url)) {
        print("The url is not valid. Please input a valid url for csv data.")
    } else {
        # download data and store in designated file when file path is valid
        if (file.exists(opt$output)) {
            download.file(opt$url, opt$output)
            cat("Dataset successfully saved in", opt$output)
        } else {
            # generate error message if the designated file path does not exist!
            print("The file path does not exist. Please input a valid file path.")
        }
    }
}

# call the main function when script is executed
main(opt)

File: src/download_data.py
# download_data.py
#
# author: UBC MDS Group 34
# date: 2020-11-27
#
# This script downloads the dataset, stores it in the designated location, and handles errors if the input url is not valid
#
# This script takes 2 arguments: the url and the output file path.
#
# Usage: Usage: src/download_data.py  --url=<url> --output_path=<output_path>

"""Downloads csv data from the web to a local filepath as a csv file format.
Usage: src/download_data.py  --url=<url> --output_path=<output_path>

Options:
--url=<url>                    URL from where to download the data (must be in standard csv format)
--output_path=<output_path>    File path including file name of where to store the file
"""

## import libraries/packages

from docopt import docopt
import requests
import os
import pandas as pd

opt = docopt(__doc__)

## define main function

def main(url, output_path):
    
# check if the URL is valid
    try: 
        request = requests.get(url)
        request.status_code == 200
        
# handle errors if the URL is not valid
    
    except Exception as req:
        print("The URL provided is not valid.")
        print(req)
    
    data = pd.read_csv(url, sep=";", header=None)
    
# save requested file to CSV locally
    
    try:
        data.to_csv(output_path, index=False)
        
# handle errors 
    
    except:
        os.makedirs(os.path.dirname(output_path))
        data.to_csv(output_path, index = False)
        
# interaction when storage is succeeded.
    
    print("File saved as", output_path)

# call the main function
    
main(opt["--url"], opt["--output_path"])